lib/kafka/consumer.rb

```diff
@@ -340,12 +340,12 @@ def fetch_batches(min_bytes:, max_wait_time:)
             @logger.warn "Partition #{topic}/#{partition} is currently paused, skipping"
           else
             @logger.debug "Fetching batch from #{topic}/#{partition} starting at offset #{offset}"
-            operation.fetch_from_partition(topic, partition, offset: offset, max_bytes: max_bytes)
+            @operation.fetch_from_partition(topic, partition, offset: offset, max_bytes: max_bytes)
           end
         end
       end
 
-      operation.execute
+      @operation.execute
```

Comment:
nope, having the operation live outside the scope of the single call is utterly key; you want to return data to the client as soon as it's available, meaning the other requests need to live on in the background.

I was shooting for "minimum lines of churn" here, just to show you the idea/approach without going full-bore clean-code; in a real PR we'd probably have to move min_bytes and max_wait_time to the initializer.